### PR TITLE
Add support for non-W3C/IETF CIB DRS URIs BT-4 BT-5 BT-9 CROM-6642

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: test/common/bond_test.js [hunk 1]
```diff
@@ -31,6 +31,14 @@ test('determineBondProvider should be "fence" if the URL host is "dg.712C"', (t)
     t.is(determineBondProvider('drs://dg.712C/anything'), BondProviders.FENCE);
 });
 
+test('determineBondProvider should be "dcf-fence" if the URL host is "dg.4DFC"', (t) => {
+    t.is(determineBondProvider('drs://dg.4DFC/anything'), BondProviders.DCF_FENCE);
+});
+
+test('determineBondProvider should be "dcf-fence" if the URL host is "dg.F82A1A"', (t) => {
```
Comment (kshakir) [inline test/common/bond_test.js]: Fix this test name heading

File: test/common/helpers.test.js [hunk 2]
```diff
@@ -65,6 +83,13 @@ test('dataObjectUriToHttps should throw an error when given a "dg.*" host with n
         'Should have thrown error but didnt!'
     );
 });
+
+test('dataObjectUriToHttps should parse "drs://dg.ANV0" Data Object uri', (t) => {
+    t.is(
+        dataObjectUriToHttps('drs://dg.ANV0/bar'),
```
Comment (kshakir) [inline test/common/helpers.test.js:89]: ~Leave a comment that this isn't used in production anymore~

Edit: Whoops. According to the [spec-beyond-the-spec](https://docs.google.com/document/d/1Wf4enSGOEXD5_AE-uzLoYqjIp5MnePbZ6kYTVFp1WoM/edit) `dg.ANV0` expanded HTTPS URL is supposed to include `dg.ANV0` in the suffix. So this test expectation on the next line is actually valid.

File: README.md [hunk 3]
```diff
@@ -131,7 +131,8 @@ Martha's `martha_v3` implementation translates requests-to and responses-from th
     - Requires OAuth for metadata: 🔐 Yes
     - Example: `drs://jade.datarepo-dev.broadinstitute.org/v1_0c86170e-312d-4b39-a0a4-2a2bfaa24c7a_c0e40912-8b14-43f6-9a2f-b278144d0060`
 - ❌ [DataGuids.org](https://dataguids.org/)
-    (any drs://dg.* other than drs://dg.4503, drs://dg.712C, drs://dg.ANV0, and not drs://dataguids.org)
+    (any drs://dg.* other than drs://dg.4503, drs://dg.712C, drs://dg.ANV0, dg.4DFC, drs://dg.F82A1A,
```
Comment (mcovarr) [inline README.md]: TOL should this be
```suggestion
    (any drs://dg.* other than drs://dg.4503, drs://dg.712C, drs://dg.ANV0, drs://dg.4DFC, drs://dg.F82A1A,
```

**Comment by kshakir** [comment]
This PR is now is ready for an official review @breilly2 @grsterin @mcovarr

File: test/common/helpers.test.js [hunk 4]
```diff
@@ -342,3 +367,101 @@ test('convertToMarthaV3Response should return null for googleServiceAccount if b
 
     t.deepEqual(convertToMarthaV3Response(mockDrsResponse), expectedResponse);
 });
+
+function testFailureResponse(t, res, expectedStatus, expectedText) {
+    t.true(Object.prototype.isPrototypeOf.call(FailureResponse.prototype, res.send.firstCall.firstArg));
```
Comment (kshakir) [inline test/common/helpers.test.js:372]: FYI @breilly2 googling the eslint error clarified the JS syntax I was supposed to be using:
- https://eslint.org/docs/rules/no-prototype-builtins
- https://stackoverflow.com/questions/18343545/javascript-isprototypeof-vs-instanceof-usage#comment26927518_18343687

File: common/helpers.js [hunk 5]
```diff
@@ -362,14 +362,14 @@ function getGsUrlFromDrsObject(drsResponse) {
...
- * @returns {MarthaV3Response} The drs object converted to a martha file info response
+ * @returns {MarthaV3Response} The drs object converted to a martha_v3 response
  */
 function convertToMarthaV3Response(drsResponse, bondProvider, googleSA) {
     const {
         checksums,
-        created_time: createdTime,
         mime_type: mimeType = 'application/octet-stream',
         size: maybeNumberSize,
+        created_time: createdTime,
```
Comment (mcovarr) [inline common/helpers.js:372]: TOL is there a reason this was moved?

File: martha/martha_v3.js [hunk 6]
```diff
@@ -87,30 +94,138 @@ class DrsType {
...
+     */
+
+    /*
+    The many, many forms of Compact Identifier-based (CIB) DRS URIs to W3C/IETF HTTPS URL conversion:
+    - https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.1.0/docs/#_compact_identifier_based_drs_uris
+    - https://docs.google.com/document/d/1Wf4enSGOEXD5_AE-uzLoYqjIp5MnePbZ6kYTVFp1WoM/edit
+    - https://broadworkbench.atlassian.net/browse/BT-4?focusedCommentId=35980
+    - etc.
+
+    Note: GA4GH CIB URIs are are incompatible with W3C/IETF URIs and the various standard libraries that parse them:
```
Comment (mcovarr) [inline martha/martha_v3.js]:
```suggestion
    Note: GA4GH CIB URIs are incompatible with W3C/IETF URIs and the various standard libraries that parse them:
```

**Review by mcovarr — COMMENTED** [review]
first pass

File: common/helpers.js (hunk 5 above)
Comment (kshakir) [reply common/helpers.js:372]: I wanted the times closer together. Happy to move it back. 🤷‍♀️

Comment (mcovarr) [reply common/helpers.js:372]: It's 💯  like this, I was just curious if there was a deeper meaning I had missed. 🙂

**Review by grsterin — APPROVED** [review]

**Review by mcovarr — APPROVED** [review]

**Comment by kshakir** [comment]
jenkins retest

**Pull request merged**